dd-sdk-android/src/main/kotlin/com/datadog/android/error/internal/DatadogExceptionHandler.kt

```diff
@@ -11,18 +11,25 @@ import com.datadog.android.core.internal.CoreFeature
 import com.datadog.android.core.internal.data.Writer
 import com.datadog.android.core.internal.net.info.NetworkInfoProvider
 import com.datadog.android.core.internal.utils.triggerUploadWorker
-import com.datadog.android.log.LogAttributes
 import com.datadog.android.log.internal.domain.Log
+import com.datadog.android.log.internal.domain.LogGenerator
 import com.datadog.android.log.internal.user.UserInfoProvider
 import com.datadog.android.rum.GlobalRum
 import com.datadog.android.rum.RumErrorSource
 import com.datadog.android.rum.internal.monitor.AdvancedRumMonitor
-import io.opentracing.util.GlobalTracer
 import java.lang.ref.WeakReference
 
 internal class DatadogExceptionHandler(
-    private val networkInfoProvider: NetworkInfoProvider?,
-    private val userInfoProvider: UserInfoProvider,
+    userInfoProvider: UserInfoProvider,
+    networkInfoProvider: NetworkInfoProvider,
+    private val logGenerator: LogGenerator = LogGenerator(
```

Comment:
Instead of setting a default value here, I'd rather see the LogGenerator created once (e.g.: in the `CoreFeature`)  and injected when this class is instantiated. 
Default values are a good ideas for public APIs, or when the default value is trivial (empty string, empty list, NoOpImpl, …), but here I don't think it's necessary.

Reply:
You cannot create it only once because each Logger can have its own custom serviceName, loggerName. We could eventually increate the number of parameters of that `generateLog` method but they're already too many I am afraid.

Reply:
We could inject it in the constructor though but we cannot keep only a singleton instance IMO

Reply:
Yes I guess you're right but at least indeed inject it from the code calling the constructor.

Reply:
Yup done.